includes/Core/Authentication/Helpers/Block_External.php

```diff
@@ -0,0 +1,50 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\Authentication\Helpers\Block_External
+ *
+ * @package   Google\Site_Kit
+ * @copyright 2025 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Authentication\Helpers;
+
+use Google\Site_Kit\Core\Util\URL;
+use Google\Site_Kit_Dependencies\GuzzleHttp\Exception\RequestException;
+use WP_Http;
+
+/**
+ * Class for blocking external requests.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class Block_External {
```

Comment:
This doesn't relate to `Authentication` specifically. I was thinking we'd model this after https://github.com/guzzle/guzzle/blob/7.9.2/src/Middleware.php

Let's rename this to `Google\Site_Kit\Core\HTTP\Middleware` and keep the static method as the way to get the middleware as it is now.